core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/stringfunction/LpadExpression.kt

```diff
@@ -4,34 +4,43 @@ import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.toNumberType
 import ch.ergon.dope.resolvable.expression.unaliased.type.toStringType
 import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.NumberType
 import ch.ergon.dope.validtype.StringType
 
 class LpadExpression(
     private val inStr: TypeExpression<StringType>,
-    private val size: Int,
+    private val size: TypeExpression<NumberType>,
     private val extra: TypeExpression<StringType> = " ".toStringType(),
 ) : TypeExpression<StringType>, FunctionOperator {
-    override fun toQueryString(): String =
-        if (extra.toQueryString().isBlank()) {
-            toFunctionQueryString(symbol = "LPAD", inStr, size.toNumberType())
-        } else {
-            toFunctionQueryString(symbol = "LPAD", inStr, size.toNumberType(), extra)
-        }
+    override fun toQueryString(): String = toFunctionQueryString(symbol = "LPAD", inStr, size, extra = extra)
 }
 
-fun lpad(inStr: TypeExpression<StringType>, size: Int, extra: TypeExpression<StringType> = "".toStringType()) =
+fun lpad(
+    inStr: TypeExpression<StringType>,
+    size: TypeExpression<NumberType>,
+    extra: TypeExpression<StringType> = " ".toStringType(),
```

Comment:
what do you mean with `extra`? And why is there a space as a default value?

Reply:
https://docs.couchbase.com/cloud/n1ql/n1ql-language-reference/stringfun.html#fn-str-lpad
LPad has a thir parameter which is optional. The Third Param (extra) in Couchbase gets added to the left of the inStr if the size is bigger than the inStr length. And the extra is per default a space.

Reply:
Can you rename 'extra' to something more meaningful (what it's actually used for) and maybe make it nullable if its optional

Reply:
done